rfcs/require-by-string.md

```diff
@@ -0,0 +1,349 @@
+
+- Libraries are fully compatible with the Roblox engine, as require-by-string is currently unimplemented.
+- Luau currently implements relative paths in relation to the current working directory. 
+- We propose changing this behavior and breaking backwards compatibility on this front.
+- With the current implementation, requiring a library that itself contains relative-path require statements [can become a mess](https://github.com/Roblox/luau/issues/959) if the Luau VM is not launched from the "correct" working directory.
+- We propose the following change: relative paths passed to require statements will be evaluated in relation to the requiring file's location, not in relation to the current working directory.
+    - Caveat 1: if the relative-path require is executed directly in a REPL input prompt (not in a file), it will be evaluated relative to the pseudo-file `stdin`, located in the current working directory.
+    - Caveat 2: if the code being executed is not tied to a file (e.g. using `loadstring`), executing require statements contained in this code will throw an error.
+- If this causes issues, we can later introduce a default alias for the current working directory (e.g. `@CWD`).
+
+## Alternatives
+
+### Different ways of defining aliases
+
+#### Defining paths/aliases directly in the requiring file
+
+Rather than defining paths/alias maps in an external configuration file, we could alternatively define paths/aliases directly in the files that require them. For example, this could manifest itself through an extension of the `--!` comment syntax or introduce new syntax like `--@<ALIAS> = @<PATH>`.
+```lua
+--@"Roact" = @"C:/LuauModules/Roact-v1.4.2"
+local Roact = require("@Roact")
+
+-- Same as this:
+local Roact = require("@C:/LuauModules/Roact-v1.4.2")
+```
+
+Some potential issues with this approach:
+- Could lead to Luau file headers becoming cluttered.
+- Would probably lead to substantial copy-and-pasting between modules in the same library, as they would likely need to share certain paths/aliases.
+- Using configuration files for paths/alias maps allows modules to share aliases while still providing the flexibility to override if needed. This approach does not support inheritance and overriding in an obvious way.
+- Removes the layer of abstraction that is provided by external paths/alias maps. This might also blur the scope of package managers. The software would need to directly modify lines of code in `.luau` files, rather than modifying configuration files.
+
+#### Defining configuration files in a non-JSON format
+
+Alias maps could alternatively be defined in specially named `.luau` files themselves, or at least adhering to Luau syntax. This approach is somewhat unappealing, however, as it would require package management software to parse Luau syntax. JSON syntax is well-understood and well-supported, which would likely faciliate the development of package management software.
```

Comment:
```suggestion
Alias maps could alternatively be defined in specially named `.luau` files themselves, or at least adhering to Luau syntax. This approach is somewhat unappealing, however, as it would require package management software to parse Luau syntax. JSON syntax is well-understood and well-supported, which would likely facilitate the development of package management software.
```